qutip/visualization.py

```diff
@@ -1213,59 +1235,55 @@ def plot_expectation_values(results, ylabels=[], title=None, show_legend=False,
     title : string
         The title of the figure.
 
-    show_legend : bool
-        Whether or not to show the legend.
-
     fig : a matplotlib Figure instance
         The Figure canvas in which the plot will be drawn.
 
-    axes : a matplotlib axes instance
+    axes : (list of)  axes instances
         The axes context in which the plot will be drawn.
 
-    figsize : (width, height)
-        The size of the matplotlib figure (in inches) if it is to be created
-        (that is, if no 'fig' and 'ax' arguments are passed).
-
     Returns
     -------
-    fig, ax : tuple
-        A tuple of the matplotlib figure and axes instances used to produce
-        the figure.
+    fig, axes : tuple
+        A tuple of the matplotlib figure and array of axes instances
+        used to produce the figure.
     """
     if not isinstance(results, list):
         results = [results]
 
     n_e_ops = max([len(result.expect) for result in results])
 
-    if not fig or not axes:
-        if not figsize:
-            figsize = (12, 3 * n_e_ops)
-        fig, axes = plt.subplots(n_e_ops, 1, sharex=True,
-                                 figsize=figsize, squeeze=False)
+    if axes is None:
+        if fig is None:
+            fig = plt.figure()
+        axes = np.array([fig.add_subplot(n_e_ops, 1, i+1)
+                         for i in range(n_e_ops)])
+
+    # create np.ndarray if axes is one axes object or list
+    if not isinstance(axes, np.ndarray):
+        if not isinstance(axes, list):
+            axes = [axes]
+        axes = np.array(axes)
 
     for r_idx, result in enumerate(results):
         for e_idx, e in enumerate(result.expect):
-            axes[e_idx, 0].plot(result.times, e,
-                                label="%s [%d]" % (result.solver, e_idx))
+            axes[e_idx].plot(result.times, e,
+                             label="%s [%d]" % (result.solver, e_idx))
 
     if title:
         fig.suptitle(title)
 
-    axes[n_e_ops - 1, 0].set_xlabel("time", fontsize=12)
+    axes[n_e_ops - 1].set_xlabel("time", fontsize=12)
     for n in range(n_e_ops):
-        if show_legend:
-            axes[n, 0].legend()
         if ylabels:
-            axes[n, 0].set_ylabel(ylabels[n], fontsize=12)
+            axes[n].set_ylabel(ylabels[n], fontsize=12)
 
     return fig, axes
 
 
-def plot_spin_distribution_2d(P, THETA, PHI,
-                              fig=None, ax=None, figsize=(8, 8)):
+def plot_spin_distribution(P, THETA, PHI, projection='2d', *,
+                           cmap=None, colorbar=False, fig=None, ax=None):
```

Comment:
These `THETA` and `PHI` are  not the same as the ones from `sphereplot` (they need to be 2d), but have the same range limit.